python-flask/python-flask-example.py

```diff
@@ -0,0 +1,35 @@
+from flask import Flask, json, request
+import base64
+import pprint
+import datetime
+app = Flask(__name__)
+
+@app.route('/', defaults={'path': ''}, methods=['POST'])
+@app.route('/<path:path>', methods=['POST'])
+def log_payload(path):
+    # Get request payload
+    payload = request.json
+
+    # Ensure that payload is a dictionary
+    if not isinstance(payload, dict):
+        payload = {'raw_data':payload}
+
+    # Get message data
+    timestamp = datetime.datetime.now()
+    message = payload.get('message',{})
+    if message:
+        # Get device ID
+        id = message.get('attributes',{}).get('deviceId','')
+        # Get base64 encoded data payload
+        data = base64.b64decode(message.get('data',''))
+        print("----- Got message from {} ({}) -------".format(id, timestamp))
+        print("Message path: /{}".format(path))
+        pprint.pprint(json.loads(data))
+    else:
+        print("----- Got message that could not be parsed ({}) -----".format(timestamp))
+        print("Message path: /{}".format(path))
+        print(payload)
+    return request.json
```

Comment:
Out of own curiosity, what exactly does this return code do?

Reply:
This just returns the payload as payload in the response, it is a bit vague, so added some real error codes if the request is non valid

Reply:
Changed it to not return the payload, just the status code and some text